yellowbrick/cluster/elbow.py

```diff
@@ -260,7 +293,25 @@ def fit(self, X, y=None, **kwargs):
             self.k_timers_.append(time.time() - start)
             self.k_scores_.append(
                 self.scoring_metric(X, self.estimator.labels_)
-            )
+            )    
+
+        if self.locate_elbow:
+            if self.metric == 'distortion':
+                self.curve_nature = 'convex'
+                self.curve_direction = 'decreasing'
+            elif self.metric=='silhouette' or self.metric=='calinski_harabaz':
+                self.curve_nature = 'concave'
+                self.curve_direction = 'increasing'    
```

Comment:
This is not a big deal and won't block merging of this code, but I wanted to make a recommendation because I'm hoping you will be a regular contributor to Yellowbrick, and this is both a code style thing and an user-facing API thing. 

I mentioned that learned attributes, suffixed with an underscore, are only set in `fit()` (nice work with those attrs, by the way!) In YB/sklearn terms, the other type of attributes are _hyperparameters_ - the properties that the user passes into `__init__` - unlike other python programming, sklearn and YB treat these specially. All other attributes of the class should be methods, e.g. `fit()` or they should be marked as private, e.g. prefixed with an `_`. 

Based on this, technically (again not a big deal) curve direction and nature should be `self._curve_direction`  and `self._curve_nature`.  

There is another little item here, though, and that's that the curve nature and direction should depend only on the metric. What happens if the user changes these values? (nothing in the case of your code) It could become confusing for the user. 

Therefore I would propose that we simply keep these as local variables when instantiating the `KneeLocator`. Simply removing the `self.` is fine - but I also wanted to show you how you might see this in other YB code:

```python
locator_kwargs = {
     'distortion': {'curve_nature': 'convex', 'curve_direction': 'decreasing'},
     'silhouette': {'curve_nature': 'concave', 'curve_direction': 'increasing'},
     'calinski_harabaz': {'curve_nature': 'concave', 'curve_direction': 'increasing'},
}.get(self.metric, {})

elbow_locator = KneeLoator(self.k_values_, self.k_scores, **locator_kwargs)
```

This is a jump table pattern and we often use it over if/elif so that it's easy to make modifications on a per-metric basis.

Reply:
This jump table pattern is excellent. I always wondered what `**kwargs` used for , today I learnt how they can be used. It's lot better over the if/else , looks tidy too.